sporkfish/endgame_tablebase.py

```diff
@@ -0,0 +1,123 @@
+import logging
+import os
+import sys
+from typing import Optional
+
+import chess
+import chess.syzygy
+
+from .board.board import Board
+from .configurable import Configurable
+
+
+class EndgameTablebaseConfig(Configurable):
+    """Configuration class for an endgame tablebase.
+
+    Attributes:
+        endgame_tablebase_path (Optional[str]): Relative (to root directory) or absolute path to endgame tablebase binary. Defaults to None.
+
+    """
+
+    def __init__(self, endgame_tablebase_path: Optional[str] = None):
+        """
+        Initialize an EndgameTablebaseConfig instance.
+
+        Args:
+            endgame_tablebase_path (Optional[str]): Relative (to root directory) or absolute path to endgame tablebase binary. Defaults to None.
+        """
+        self.endgame_tablebase_path = endgame_tablebase_path
+
+
+class EndgameTablebase:
+    """Class for handling the endgame tablebase in chess engines."""
+
+    def __init__(
+        self, config: EndgameTablebaseConfig = EndgameTablebaseConfig()
+    ) -> None:
+        """
+        Initialize the EndgameTablebase instance.
+
+        :param config: Endgame tablebase config. If not provided, the default endgame tablebase is used.
+        :type config: EndgameTablebaseConfig
+        :return: None
+        """
+        self._config = config
+        if self._config.endgame_tablebase_path:
+            self._db = self._load(
+                self._resource_path(self._config.endgame_tablebase_path)
+            )
+        else:
+            logging.warning(
+                f"Skip loading endgame tablebase as the endgame tablebase binary path is not passed in configuration."
+            )
+
+    def _resource_path(self, relative_path: str) -> str:
+        """
+        Get the absolute path to a resource.
+
+        :param relative_path: Relative path to the resource.
+        :type relative_path: str
+        :return: Absolute path to the resource.
+        :rtype: str
+        """
+        # PyInstaller creates a temp folder and stores path in _MEIPASS
+        if hasattr(sys, "_MEIPASS"):
+            base_path = sys._MEIPASS
+        else:
+            base_path = os.path.abspath(".")
+        return os.path.join(base_path, relative_path)
```

Comment:
WRT relative path: I believe so, this is because I still want to keep PyInstaller support to be able to freeze the program in to an executable. I see most elo estimation websites / even old lichess-bot require a compiled binary / exe of the engine, rather than the python program.

Passing the decision of renaming to @michelle-lai-swl

Reply:
Fixed, thanks!